### PR TITLE
Refactor Submission panel part2

### PR DESCRIPTION
**Pull request by wezhang**

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/ui/SparkSubmissionContentPanelConfigurable.java [hunk 1]
```diff
@@ -1,387 +0,0 @@
...
-        data.setReferenceFiles(referencedFileList);
-        data.setReferenceJars(uploadedFilePathList);
-        data.setCommandLineArgs(argsList);
-
-        data.setTableModel(tableModel);
-        data.setClusterComboBoxModel(submissionPanel.getClustersModel());
-
-        // get Job upload storage panel data
-        getStorageWithUploadPathPanel().getData(data.getJobUploadStorageModel());
-        data.getErrors().add(jobUploadStorageCtrl.getResultMessage());
```
Comment (wezhang) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/ui/SparkSubmissionContentPanelConfigurable.java]: Should be a regression here for error checking

**Comment by wezhang** [comment]
```
+ diff /home/jenkins/dependencyLog//develop/intellij/910-intellij.log /home/jenkins/dependencyLog//PR/intellij/2529-intellij.log
+ cat
+ cat /home/jenkins/dependencyLog//patch/PR-2529.patch
+ wc -l
+ [ 12 -eq 0 ]
+ cat /home/jenkins/dependencyLog//patch/PR-2529.patch
25c25
< |    +--- com.microsoft.azure:applicationinsights-core:[2.0.0, 3.0.0) -> 2.2.1
---
> |    +--- com.microsoft.azure:applicationinsights-core:[2.0.0, 3.0.0) -> 2.3.0
27c27
< |    +--- com.microsoft.azure:applicationinsights-web:[2.0.0, 3.0.0) -> 2.2.1
---
> |    +--- com.microsoft.azure:applicationinsights-web:[2.0.0, 3.0.0) -> 2.3.0
39c39
< |    +--- com.microsoft.azure:applicationinsights-core:[2.0.0, 3.0.0) -> 2.2.1
---
> |    +--- com.microsoft.azure:applicationinsights-core:[2.0.0, 3.0.0) -> 2.3.0
+ echo ALERT: New dependencies introduced!
ALERT: New dependencies introduced!
```

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/ui/CosmosSparkSubmissionPanel.kt [hunk 2]
```diff
@@ -22,40 +22,29 @@
...
-import com.microsoft.azure.hdinsight.sdk.common.azure.serverless.AzureSparkServerlessClusterManager
 import com.microsoft.azure.hdinsight.spark.common.CosmosSparkSubmitModel
 import com.microsoft.azure.hdinsight.spark.common.SparkSubmitModel
-import rx.Observable
 
-class CosmosSparkSubmissionPanelConfigurable(project: Project)
-    : SparkSubmissionContentPanelConfigurable(project), ILogger {
-    override fun getType(): String = "Azure Data Lake Spark Pool"
+class CosmosSparkSubmissionPanel(project: Project)
+    : SparkSubmissionContentPanel(project, "Azure Data Lake Spark Pool"), ILogger {
```
Comment (konjac) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/ui/CosmosSparkSubmissionPanel.kt]: ILogger is not necessary?

Comment (wezhang) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/ui/CosmosSparkSubmissionPanel.kt]: You're right! I will fix it. 😄

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/common/viewmodels/SwingComponentPropertyDelegated.kt [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    }
+
+}
+
+class ComboBoxModelDelegated<T>(private val comboBox: JComboBox<T>) {
+    operator fun getValue(ref: Any?, property: KProperty<*>): ComboBoxModel<T> {
+        return comboBox.model
+    }
+
+    operator fun setValue(ref: Any?, property: KProperty<*>, v: ComboBoxModel<T>) {
```
[the remark this answers is not in the document]
Comment (wezhang) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/common/viewmodels/SwingComponentPropertyDelegated.kt]: The combe model is already ViewModel design. We can set it without consider it's in dispatch thread or not.

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/ui/ServerlessSparkConfigurable.kt [hunk 4]
```diff
@@ -25,6 +25,6 @@ package com.microsoft.azure.hdinsight.spark.ui
 import com.intellij.openapi.project.Project
 
 class ServerlessSparkConfigurable(project: Project) : SparkBatchJobConfigurable(project) {
```
Comment (t-rufang) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/ui/ServerlessSparkConfigurable.kt]: >ServerlessSparkConfigurable [](start = 6, length = 27)

`ServerlessSparkConfigurable` should be renamed to `CosmosSparkConfigurable` ?

**Pull request merged**